File: PHP/1028.php
<?php
// Problema: 1028 - Figurinhas  | Resposta: Accepted
// Linguagem: PHP (8.1.2) [+1s] | Tempo: 0.203s

function gcd(int $a, int $b): int {
    if ($b == 0) { return $a; }
    return gcd($b, $a % $b);
}

$casos = intval(trim(fgets(STDIN)));
for ($i = 0; $i < $casos; $i++) {
    $entrada = array_map(
        'intval', 
        explode(' ', trim(fgets(STDIN)))
    );

    echo gcd($entrada[0], $entrada[1]) . PHP_EOL;
}
?>

File: C/1028.c
// Problema: 1028 - Figurinhas | Resposta: Accepted
// Linguagem: C99 [+0s]        | Tempo: 0.000s

#include <stdio.h>

int gcd(int a, int b) {
    while (b) {
        int temp = b;
        b = a % b;
        a = temp;
    }
    return a;
}

int main(){
    int rep, a, b;
    scanf("%d", &rep);
    while(rep--){
        scanf("%d %d", &a, &b);
        printf("%d\n", gcd(a, b));
    }
    return 0;
}